src/Powercord/plugins/pc-routelinks/index.js

```diff
@@ -0,0 +1,110 @@
+const { Plugin } = require('powercord/entities');
+const { getRepoInfo, cloneRepo } = require('../pc-moduleManager/util');
+const express = require('express');
+const { open: openModal, close: closeModal } = require('powercord/modal');
+const { React, getModule } = require('powercord/webpack');
+const Modal = require('./components/ConfirmModal');
+
+
+module.exports = class RDLinks extends Plugin {
+  async startPlugin () {
+    this.queue = new Set();
+    this.app = express();
+    this.showNotification = await getModule([ 'showNotification' ]);
+    this.focus = await getModule([ 'focus' ]);
+
+    this.app.use((req, res, next) => {
+      res.header('Access-Control-Allow-Origin', '*');
+      res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
+      res.header('Access-Control-Allow-Methods', 'POST');
+      next();
+    });
+
+    this.app.post('/', (req, res) => {
+      if (req.header.origin === 'replugged.dev') {
+        res.sendStatus(400);
+      }
+      // res.sendStatus(400); for debug only, commented out so no good reply is sent in case of bad origin
+    });
+
+    this.app.post('/install/', async (req, res) => {
+      if (req.header.origin !== 'replugged.dev') {
+        return;
+      }
+      this.info = await getRepoInfo(req.query.address);
+      if (this.info) {
+        if (this.info.isInstalled) {
+          res.status(403).json({ error: 'Already installed',
+            installed: true,
+            promptSent: false,
+            cannotFind: false
+          });
+          return;
+        }
+        res.status(200).json({ error: 'Success',
+          installed: false,
+          promptSent: true,
+          cannotFind: false
+        });
+        this.info.url = req.query.address;
+        this.openInstallModal();
+      } else {
+        res.status(404).json({ error: 'Cannot find repository',
+          installed: false,
+          promptSent: false,
+          cannotFind: true
+        });
+      }
```

Comment:
Update these objects to have a code (eg `ALREADY_INSTALLED`) and a plain text message.